### PR TITLE
[SPARK-36054][SQL] Support group by TimestampNTZ type column

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Support group by TimestampNTZ type column

### Why are the changes needed?

It's a basic SQL operation.

### Does this PR introduce _any_ user-facing change?

No, the new timestmap type is not released yet.

### How was this patch tested?

Unit test

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala [hunk 1]
```diff
@@ -1398,6 +1398,14 @@ class DataFrameAggregateSuite extends QueryTest
     val df2 = Seq(Period.ofYears(1)).toDF("a").groupBy("a").count()
     checkAnswer(df2, Row(Period.ofYears(1), 1))
   }
+
+  test("SPARK-36054: Support group by TimestampNTZ column") {
+    val ts1 = "2021-01-01T00:00:00"
+    val ts2 = "2021-01-01T00:00:01"
+    val localDateTime = Seq(ts1, ts1, ts2).map(LocalDateTime.parse)
+    val df = localDateTime.toDF("ts").groupBy("ts").count().orderBy("ts")
+    checkAnswer(df, Seq(Row(LocalDateTime.parse(ts1), 2), Row(LocalDateTime.parse(ts2), 1)))
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala:1407]: Could you check type of the result, please. Is it `timestamp_ntz`?

Comment (gengliangwang) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala:1410]: Done

**Review by MaxGekk — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Comment by MaxGekk** [comment]
The failed GA is related, it seems:
```
TwoLevelAggregateHashMapWithVectorizedMapSuite.SPARK-36054: Support group by TimestampNTZ column
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1252.0 failed 1 times, most recent failure: Lost task 0.0 in stage 1252.0 (TID 1028) (localhost executor driver): java.lang.RuntimeException: Unhandled TimestampNTZType
	at org.apache.spark.sql.execution.vectorized.OnHeapColumnVector.reserveInternal(OnHeapColumnVector.java:572)
```
@gengliangwang Could you fix it, please.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.2.
Thank you, @gengliangwang and @cloud-fan for review.

**Pull request closed without merging**